src/pages/docs/registration-authorities/acme-for-vault.mdx

```diff
@@ -0,0 +1,266 @@
+To create these, run:
+
+```bash
+vault write pki_int1/roles/ec-role key_type=ec allow_any_name=true enforce_hostnames=false max_ttl=72h
+vault write pki_int1/roles/rsa-role key_type=rsa allow_any_name=true enforce_hostnames=false max_ttl=72h
+vault write pki_int1/roles/ed25519-role key_type=ed25519 allow_any_name=true enforce_hostnames=false max_ttl=72h
+```
+
+You'll also need policies for each role. Run:
+
+```bash
+echo 'path "pki_int1/sign/ec-role" {
+    capabilities = ["create", "update"]
+}
+path "pki_int1/sign/rsa-role" {
+    capabilities = ["create", "update"]
+}
+path "pki_int1/sign/ed25519-role" {
+    capabilities = ["create", "update"]
+}
+path "pki_int/revoke" {
+    capabilities = ["create", "update"]
+}' > policy.hcl
+vault policy write smallstep policy.hcl
+```
+
+### 4. Enable Vault authentication for your RA.
+
+For the last part of our Vault setup, we need to configure Vault authentication.
+Run:
+
+```bash
+vault auth enable -description="SmallStep RA AppRole" approle
+vault write auth/approle/role/approle \
+     bind_secret_id=true \
+     local_secret_ids=false \
+     secret_id_num_uses=0 \
+     secret_id_ttl=0s \
+     token_explicit_max_ttl=0s \
+     token_max_ttl=0s \
+     token_no_default_policy=false \
+     token_num_uses=0 \
+     token_period=0s \
+     token_policies="smallstep" \
+     token_ttl=0s \
+     token_type=default
+VAULT_ROLE_ID=$(vault read -field=role_id auth/approle/role/approle/role-id)
+VAULT_SECRET_ID=$(vault write -field=secret_id -f auth/approle/role/approle/secret-id)
+```
+
+We'll need the `VAULT_ROLE_ID` and `VAULT_SECRET_ID` to configure our RA in the next step.
+
+### 5. Configure the RA server.
+
+In this section, we'll initialize a basic CA,
+and then convert it into an RA.
+
+First, run `step ca init` as described in [Getting Started](/docs/step-ca/getting-started).
+Configure the DNS names, address, and JWK provisioner password.
+
+Once the PKI is initialized, let's update `ca.json` to enable RA functionality:
+
+```bash
+VAULT_URI=http://127.0.0.1:8200
+VAULT_CA_FINGERPRINT=$(step certificate fingerprint root_ca.crt)
+cat <<< $(jq --arg vault_role_id "$VAULT_ROLE_ID" \
+            --arg vault_secret_id "$VAULT_SECRET_ID" \
+            --arg vault_uri "$VAULT_URI" \
+            --arg vault_ca_fingerprint "$VAULT_CA_FINGERPRINT" \
+             '(.authority += {
+                "type": "vaultcas",
+                "certificateAuthority": $vault_uri,
+                "certificateAuthorityFingerprint": $vault_ca_fingerprint,
+                "config": {
+                          "pki": "pki_int1",
+                          "pkiRoleRSA": "rsa-role",
+                          "pkiRoleEC": "ec-role",
+                          "pkiRoleEd25519": "ed25519-role",
+                          "appRole": "approle",
+                          "roleID": $vault_role_id,
+                          "secretID": {
+                                "fromString": $vault_secret_id
```

Comment:
You can use `fromString`, `fromEnv` or `fromFile`

Reply:
Good to know

Reply:
I've noted this and suggested ppl use `fromFile` in prod